### PR TITLE
fix(realtime): remove unnecessary check in `removeChannel`

### PR DESCRIPTION
**Pull request by snickerdoodle2**
## 🔍 Description

#2091 introduced manual channel removing in `removeChannel`, but similar behavior is already present in `RealtimeChannel`.

constructor:
```js
this._onClose(() => {
  this.rejoinTimer.reset()
  this.socket.log('channel', `close ${this.topic} ${this._joinRef()}`)
  this.state = CHANNEL_STATES.closed
  this.socket._remove(this)
})
```

`unsubscribe` method:
```js
const onClose = () => {
  this.socket.log('channel', `leave ${this.topic}`)
  this._trigger(CHANNEL_EVENTS.close, 'leave', this._joinRef())
}

// cut...
leavePush
  .receive('ok', () => {
    onClose()
    resolve('ok')
  })
  .receive('timeout', () => {
    onClose()
    resolve('timed out')
  })
  .receive('error', () => {
    resolve('error')
  })
```

### What changed?
Removed redundant check in `removeChannel`, fix tests (as they were mocking unsubscribe method, which did the removing).

## 🔄 Breaking changes

- [x] This PR contains no breaking changes

## 📋 Checklist
- [x] I have read the [Contributing Guidelines](https://github.com/supabase/supabase-js/blob/master/CONTRIBUTING.md)
- [x] My PR title follows the [conventional commit format](https://www.conventionalcommits.org/): `<type>(<scope>): <description>`
- [x] I have run `npx nx format` to ensure consistent code formatting

## Summary by CodeRabbit

* **Bug Fixes**
  * Adjusted real-time client channel removal behavior so internal channel state remains consistent when an unsubscribe attempt doesn't confirm, while preserving the reported unsubscribe status.
* **Tests**
  * Consolidated and parameterized channel-unsubscribe tests to cover multiple unsubscribe outcomes and validate the updated behavior.

### CODE REVIEW

**Review by grdsdev — APPROVED** [review]

File: packages/core/realtime-js/test/RealtimeClient.channels.test.ts [hunk 1]
```diff
@@ -154,13 +154,13 @@ describe('channel', () => {
...
-    // Mock unsubscribe to return 'ok'
-    vi.spyOn(channel, 'unsubscribe').mockResolvedValue('ok')
+    channel._onMessage = () => ({ status: 'ok' })
 
-    await testSetup.socket.removeChannel(channel)
+    const result = await testSetup.socket.removeChannel(channel)
 
     // Channel should be removed from the list
     assert.equal(testSetup.socket.getChannels().length, 0)
+    assert.equal(result, 'ok')
```
Comment (edgurgel) [inline packages/core/realtime-js/test/RealtimeClient.channels.test.ts]: Could we add a test to ensure that the list of channels is empty after this then?

Comment (snickerdoodle2) [reply packages/core/realtime-js/test/RealtimeClient.channels.test.ts]: hmm, this check is line above, unless you mean something else?

**Comment by snickerdoodle2** [comment]
I've added test case for `timeout` code path and merged all those tests into one `test.each`. tbh I don't like the name of that test, but I can't come up with anything better :/
